ddp-workspace/projects/ddp-dsm-ui/src/app/scan/scan.component.ts

```diff
@@ -84,53 +85,71 @@ export class ScanComponent implements OnInit {
     }
   }
 
-  private checkIfKitLabelChanged(left: string, right: string, position: number): boolean {
-    for (let i = 0; i < this.scanPairsValue.length; i++) {
-      if (this.scanPairsValue[i].leftValue === left && i === position) {
-        this.scanPairsValue[i].rightValue = right;
-        return true;
-      }
+  private shouldAddNewPair(position: number): boolean {
+    if(this.scanPairs.length === position + 1) {
+      return true;
     }
     return false;
   }
 
   public validateRightValue(position: number): boolean {
-    if (this.scanPairsValue.length > 0 && this.scanPairsValue[position] != null) {
-      return this.validateValue(this.scanPairsValue[position].rightValue, position, false);
+    if (this.scanPairsValue.length > 1 && this.scanPairsValue[position] != null) {
+      return this.isDuplicate(this.scanPairsValue[position].rightValue, position, false);
     }
     return false;
   }
 
   public validateLeftValue(position: number): boolean {
-    if (this.scanPairsValue.length > 0 && this.scanPairsValue[position] != null) {
-      return this.validateValue(this.scanPairsValue[position].leftValue, position, true);
+    if (this.scanPairsValue.length > 1 && this.scanPairsValue[position] != null) {
+      return this.isDuplicate(this.scanPairsValue[position].leftValue, position, true);
     }
     return false;
   }
 
-  private validateValue(labelValue: string, position: number, isLeft: boolean): boolean {
+  private isDuplicate(labelValue: string, position: number, isLeft: boolean): boolean {
     let isDuplicate = false;
-    for (let i = 0; i < this.scanPairsValue.length; i++) {
-      if (i !== position) {
-        if (this.scanPairsValue[position].leftValue != null && labelValue === this.scanPairsValue[i].leftValue
-          || this.scanPairsValue[position].rightValue != null && labelValue === this.scanPairsValue[i].rightValue) {
-          isDuplicate = true;
-        }
-      } else {
-        if (isLeft) {
-          if (this.scanPairsValue[position].rightValue != null && labelValue === this.scanPairsValue[i].rightValue) {
+    console.log("here");
```

Comment:
I think you forgot to remove console log from here

Reply:
My bad! Removed